tests.py

```diff
@@ -217,180 +220,130 @@ def test_parse_output_str__unknown_data_format(self):
         with self.assertRaises(ParsingError):
             output_type, data_format, output_url = parse_output_str(output_str)
 
-class TestInputOutput(unittest.TestCase):
-
-    def test_csv_file_in_no_out(self):
-        global LOGGING_LEVEL
-        global TESTING_IN_FILE_CSV
+    def setUp(self):
         logging.basicConfig(format='%(asctime)s %(message)s', level=LOGGING_LEVEL)
 
+    def tearDown(self):
+        pass
+
+class TestFileIO(unittest.TestCase):
+
+    DEFAULT_SLEEPING_DURATION = 2
+
+    def test_csv_file_in_no_out(self):
         pipeline = Pipeline()
         pipeline.start()
-
         file_source = FileSource(   filename=TESTING_IN_FILE_CSV,
                                     pipeline=pipeline)
         file_source.start()
-        time.sleep(2)
+        time.sleep(self.DEFAULT_SLEEPING_DURATION)
```

Comment:
Tests should not depend on sleep at all... There should be a method that waits for the pipeline to finish. After that wait() method returns, all files should be written, close and cleaned up.

Reply:
wait to finish is one of the next issues to work on. til then i would keep the sleep approach